source/code/include/playfab/PlayFabPlatformUtils.h

```diff
@@ -7,53 +7,125 @@
 
 namespace PlayFab
 {
-    typedef std::chrono::time_point<std::chrono::system_clock> TimePoint;
+#if defined(PLAYFAB_PLATFORM_SWITCH)
+    static_assert("You must request the Nintendo specific XPlat SDK from PlayFab support.");
+#else
+    typedef std::chrono::system_clock Clock;
+    typedef std::chrono::time_point<Clock> TimePoint;
+#endif
 
-    inline TimePoint GetTimePointNow()
+    // The primary purpose of these format strings is to communicate to and from the PlayFab server with consistent accuracy across platforms supported by this SDK
+    constexpr char TIMESTAMP_READ_FORMAT[] = "%Y-%m-%dT%T";
+    constexpr char TIMESTAMP_WRITE_FORMAT[] = "%Y-%m-%dT%H:%M:%S.000Z";
+
+    // Initialize may be required on some platforms
+    inline void InitializeClock()
+    {
+#if defined(PLAYFAB_PLATFORM_SWITCH)
+        static_assert("You must request the Nintendo specific XPlat SDK from PlayFab support.");
+#endif
+    }
+
+    // Time type conversions
+    inline time_t TimePointToTimeT(TimePoint input)
     {
-        return std::chrono::system_clock::now();
+        return Clock::to_time_t(input);
     }
 
-    inline time_t GetPlayFabTimeTNow()
+    inline TimePoint TimeTToTimePoint(time_t input)
     {
-        return time(0);
+        return Clock::from_time_t(input);
     }
 
-    inline std::string LocalTimeTToUtcString(time_t now)
+    inline tm TimeTToUtcTm(time_t input)
     {
         tm timeInfo;
+#if defined(PLAYFAB_PLATFORM_WINDOWS) || defined(PLAYFAB_PLATFORM_XBOX) || defined(PLAYFAB_PLATFORM_PLAYSTATION)
+#define gmtime_r gmtime_s
```

Comment:
this concerns me. I'm pretty sure the windows UCRT version of gmtime_s has the order of parameters reversed (struct tm first then time_t) https://docs.microsoft.com/en-us/cpp/c-runtime-library/reference/gmtime-s-gmtime32-s-gmtime64-s?view=vs-2019

the return type appears to be different as well. How was this validated for windows?

in general I don't think the using preprocessor macros to override functions is a good idea here but it's got me extra confused with the flipped parameters

Reply:
Tested on Windows vs 2017, and while I noticed the order was reversed, StackOverflow claimed it would work, and I verified that it did. I even confirmed that the struct was assigned properly, via automated tests, and manual debugging.
I was caught up in the beauty of it, but you're probably right, it's probably not safe overall.
I'll do a more normal version.

---
In reply to: [357515763](https://github.com/PlayFab/XPlatCoreTemplate/pull/65#discussion_r357515763) [](ancestors = 357515763)

Reply:
Oh, it's worse than that... The order you see here is correct for PlayStation, and wrong for Windows. I made a mistake when I was testing on Windows (I swapped it locally, but didn't copy it back to the template).
I've got a new version, but sadly it won't be as pretty as this. But it'll work on everything.

---
In reply to: [357774542](https://github.com/PlayFab/XPlatCoreTemplate/pull/65#discussion_r357774542) [](ancestors = 357774542,357515763)

Reply:
Scott, I'm marking this one resolved as I've addressed your original concern, re-tested carefully on multiple platforms, and confirmed that exactly this code works on all platforms.

---
In reply to: [357778543](https://github.com/PlayFab/XPlatCoreTemplate/pull/65#discussion_r357778543) [](ancestors = 357778543,357774542,357515763)

Reply:
Ok, for reals this time... :D

---
In reply to: [357785771](https://github.com/PlayFab/XPlatCoreTemplate/pull/65#discussion_r357785771) [](ancestors = 357785771,357778543,357774542,357515763)